homeassistant/components/sensor/speedtest.py

```diff
@@ -85,8 +86,17 @@ def update(self):
         """Get the latest data and update the states."""
         data = self.speedtest_client.data
         if data is None:
-            return
-
+            entity_id = 'sensor.speedtest_' + self._name.lower()
+            states = recorder.get_model('States')
+            last_state = recorder.execute(
```

Comment:
This will fail if the recorder has not been initialized.

Also, the initial implementation would just return which means it would keep the previous state, before the error happened.